### PR TITLE
Enabled CORS for communication between frontend and backend api

### LINKED ISSUE #31: Enable CORS in backend
### What happened?
Making a request to the backend from the frontend results in a XMLHTTPError

### How can we reproduce it?
Create a function that hits the GET /ping endpoint in the frontend

https://github.com/gin-contrib/cors

### PR DESCRIPTION
**Pull request by SAbbas2018**
## [DEV-031] Enable CORS in backend

Fixes: #31 

-   [Y] Has the code been formatted?
-   [Y] Have dependencies/packages been added?

### Changes
Enabled CORS for communication between frontend and backend API. Currently allowing http://localhost:3000, and https://dialog-gamma.vercel.app/ as the allowed origins. 

### Testing
-   [Y] Does the code have sufficient automated test coverage? If not, include a reasoning and steps required to manually QA
-   Tested multiple calls to backend via axios get request to the ping route and was returned status code 200.

### CODE REVIEW

File: backend/httpd/main.go [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
 	r := gin.Default()
+
+	// This allows for front end calls made from the origin source
+	// http://localhost:3000
+	// This will need to be changed/updated to either be "*", ie
+	// allow all origins (which could be bad for security), or
+	// the url for where ever we host our front end.
+	// All RESTful api methods are allowed
+	r.Use(cors.New(cors.Config{
+		AllowOrigins:     []string{"http://localhost:3000", "https://dialog-gamma.vercel.app/"},
```
Comment (jcserv) [inline backend/httpd/main.go:20]: Great idea, do you think we should add the "preview" deployments as well?
Taking a look at the other PRs, their urls (which are generated by vercel) are:
https://dialog-git-pgcompose.white-van.vercel.app/

https://dialog-git-dev-036.white-van.vercel.app/

https://dialog-git-dev-031-b.white-van.vercel.app/

**Review by jcserv — APPROVED** [review]

File: backend/httpd/main.go (hunk 1 above)
Comment (krishchow) [reply backend/httpd/main.go:20]: hm, adding all the domains manually seems like a bad idea. I think CORS supports regex/pattern matching, so we could match all sites under the white-van.vercel.app subdomain?

https://www.moesif.com/blog/technical/cors/Authoritative-Guide-to-CORS-Cross-Origin-Resource-Sharing-for-REST-APIs/
https://stackoverflow.com/questions/14003332/access-control-allow-origin-wildcard-subdomains-ports-and-protocols

Comment (SAbbas2018) [reply backend/httpd/main.go:20]: I have added regex to the AllowOriginFunction which checks to see if the incoming origin matches the allowed URLs. It should allow for calls from http://localhost:3000, and any of the white-van.vercel.app/ subdomains.

**Comment by SAbbas2018** [comment]
Still need more testing to see if the api calls return status 200, I used regex101, to check that the URLs we use for development should be accepted and they looked to be good, but we definitely need some more testing by making calls from the frontend.

**Review by krishchow — APPROVED** [review]
LGTM :)

**Pull request merged**